customer-chat/app.js

```diff
@@ -0,0 +1,196 @@
+    } else if (webhook_event.postback) {
+        handlePostback(sender_psid, webhook_event.postback);
+    } else if (webhook_event.referral) {
+        handleReferral(sender_psid, webhook_event.referral);
+    } else if (webhook_event.pass_thread_control) {
+        handlePass(sender_psid, webhook_event.pass_thread_control);
+    }
+}
+
+// Handle thread control passed event
+function handlePass(sender_psid, received_event) {
+    let response;
+    response = {
+        "text": "An automated agent is now helping you with your inquiry."
+    };
+    callSendAPI(sender_psid, response);
+}
+
+// Handles messages events
+function handleMessage(sender_psid, received_message) {
+    let response;
+    if (received_message && received_message.text == 'Customer Support') {
+        response = {
+            "text": "A human support agent is now helping you with your inquiry."
+        };
+        callSendAPI(sender_psid, response);
+        passThreadControl(sender_psid, page_inbox_app_id);
+    } else if (received_message.text) {
+        if ((received_message.tags) && (received_message.tags.source == 'customer_chat_plugin')) {
+            response = {
+                "text": `We hope you enjoyed our website, we see you would like help with:\n "${received_message.text}".`
+            };
+        } else {
+            response = {
+                "text": `We see you would like help with ${received_message.text}.`
+            };
+        }
+        callSendAPI(sender_psid, response);
+    }
+}
+
+// Handles messaging_postbacks events
+function handlePostback(sender_psid, received_postback) {
+    console.log(`PSID: ${sender_psid}, postback: ${received_postback.ref}`);
+}
+
+// Handles messaging_referral events
+function handleReferral(sender_psid, received_referral) {
+    console.log(`PSID: ${sender_psid}, referral: ${received_referral.ref}`);
+}
+
+// Pass thread control
+function passThreadControl(sender_psid, target_app_id) {
+    console.log('PASSING THREAD CONTROL');
+    let request_body = {
+        recipient: {
+            id: sender_psid
+        },
+        target_app_id
+    };
+
+    // Send the HTTP request to the Messenger Platform
+    request({
+        "uri": "https://graph.facebook.com/v2.6/me/pass_thread_control",
+        "qs": {"access_token": access_token},
+        "method": "POST",
+        "json": request_body
+    }, (err, res, body) => {
+        if (!err) {
+            console.log('message sent!')
+        } else {
+            console.error(`Unable to send message:${err}`);
+        }
+    });
+}
+
+// Sends response messages via the Send API
+function callSendAPI(sender_psid, response) {
+    // Construct the message body
+    let request_body = {
+        "recipient": {
+            "id": sender_psid
+        },
+        "message": response
+    };
+
+    // Send the HTTP request to the Messenger Platform
+    request({
+        "uri": "https://graph.facebook.com/v2.6/me/messages",
```

Comment:
let's remove the version from the path

Reply:
@amuramoto I've never seen an example without the version in the path, can you just remove it and it uses latest?

Reply:
@ChrisChinchilla  yup

Reply:
Actually right now it uses 2.11 I think, but normally it uses latest